cmd/s2i/main.go

```diff
@@ -356,28 +358,31 @@ func checkErr(err error) {
 
 func main() {
 	cfg := &api.Config{}
-	stiCmd := &cobra.Command{
+	s2iCmd := &cobra.Command{
 		Use: "s2i",
 		Long: "Source-to-image (S2I) is a tool for building repeatable docker images.\n\n" +
 			"A command line interface that injects and assembles source code into a docker image.\n" +
 			"Complete documentation is available at http://github.com/openshift/source-to-image",
+		PersistentPreRun: func(cmd *cobra.Command, args []string) {
+			if fileLogger, ok := glog.(*utilglog.FileLogger); ok {
+				fileLogger.SetLoggerLevel(*loglevelPtr)
+			}
+		},
```

Comment:
That makes a bit more sense, but I remain concerned on three fronts:

1. There is an override in s2i which forces glog to output to stderr. The comments around it complain that it should be redirected to stdout. https://github.com/jupierce/source-to-image/blob/master/cmd/s2i/main.go#L332-L334 . This again suggest to me that we want control of our own destiny here instead of delegating to glog. 
2. In s2i, glog's capabilities almost completely suppressed because cobra's CLI library prevents important flags from reaching glog (vmodule can't be used, for example, because it is not registered with cobra and would not be passed to glog even if it were). This is why the original code bothered blowing --loglevel into -v: https://github.com/jupierce/source-to-image/blob/master/cmd/s2i/main.go#L327-L331 .
3. utilglog interferes with glog's ability to report the correct file/line number.

Combining the above means glog is mostly useless for delegation. It can only log to stderr (which filelogger is already doing), it can't be configured from the command line, and its prefix is incorrect if delegation is triggered.